snapcraft/tests/repo/test_snaps.py

```diff
@@ -0,0 +1,377 @@
+            return 'email: -'.encode()
+
+
+class FakeSnapdServer(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        parsed_url = parse.urlparse(self.path)
+        if parsed_url.path.startswith('/v2/snaps/'):
+            self._handle_snaps(parsed_url)
+        elif parsed_url.path == '/v2/find':
+            self._handle_find(parsed_url)
+        else:
+            self.wfile.write(parsed_url.path.encode())
+
+    def _handle_snaps(self, parsed_url):
+        status_code = 404
+        params = {}
+
+        if parsed_url.path.endswith('/fake-snap'):
+            status_code = 200
+            params = {'channel': 'stable'}
+        elif parsed_url.path.endswith('/fake-snap-stable'):
+            status_code = 200
+            params = {'channel': 'stable'}
+        elif parsed_url.path.endswith('/fake-snap-branch'):
+            status_code = 200
+            params = {'channel': 'candidate/branch'}
+        elif parsed_url.path.endswith('/fake-snap-track-stable'):
+            status_code = 200
+            params = {'channel': 'track/stable'}
+        elif parsed_url.path.endswith('/fake-snap-track-stable-branch'):
+            status_code = 200
+            params = {'channel': 'track/stable/branch'}
+        elif parsed_url.path.endswith('/fake-snap-edge'):
+            status_code = 200
+            params = {'channel': 'edge'}
+
+        self.send_response(status_code)
+        self.send_header('Content-Type', 'text/application+json')
+        self.end_headers()
+        response = json.dumps({'result': params}).encode()
+        self.wfile.write(response)
+
+    def _handle_find(self, parsed_url):
+        query = parse.parse_qs(parsed_url.query)
+        status_code = 404
+        params = {}
+
+        if query['name'][0] == 'fake-snap':
+            status_code = 200
+            params = {'channels': {
+                'latest/stable': {'confinement': 'strict'},
+                'classic/stable': {'confinement': 'classic'},
+                'strict/stable': {'confinement': 'strict'},
+                'devmode/stable': {'confinement': 'devmode'},
+            }}
+        if query['name'][0] == 'new-fake-snap':
+            status_code = 200
+            params = {'channels': {
+                'latest/stable': {'confinement': 'strict'},
+            }}
+
+        self.send_response(status_code)
+        self.send_header('Content-Type', 'text/application+json')
+        self.end_headers()
+        response = json.dumps({'result': [params]}).encode()
+        self.wfile.write(response)
+
+
+class UnixHTTPServer(socketserver.UnixStreamServer):
+
+    def get_request(self):
+        request, client_address = self.socket.accept()
+        # BaseHTTPRequestHandler expects a tuple with the client address at
+        # index 0, so we fake one
+        if len(client_address) == 0:
+            client_address = (self.server_address,)
+        return (request, client_address)
+
+
+class FakeSnapd(fixtures.Fixture):
```

Comment:
I like your new FakeSnapd a lot. But we already have a FakeSnapd fixture in `snapcraft/tests/fixture_setup.py` so please merge the two.

Reply:
I would rather not do it here as it would require substantial changes. I'll do it in a follow up PR though.